### ISSUE TITLE
Internal Server Error if request to Rollbar fails

### ISSUE THREAD

**Issue by ktalebian** [Scalingo/lua-resty-rollbar#2]
The request to nginx fails with `Internal Server Error` if the request to rollbar fails. For example, set an incorrect token, and then use the reporter. You'll get a 

```
[error] 1480#0: *69 [lua] rollbar.lua:100: invalid Rollbar response: 401 {
   "err": 1,
   "message": "invalid access token"
}, context: ngx.timer, client: x.x.x.x, server: 0.0.0.0:80
```

And the request fails. Failure to publish to rollbar should not cause the service to fail and the failure should be handled gracefully.

**Comment by EtienneM** [comment]
You're right. I'll not have time to work on it right away but if you can submit a PR on this one, I will happily merge it in master.

**Comment by EtienneM** [comment]
@ktalebian I will actually need more details on this issue. The Rollbar request is done in a light thread:

https://github.com/Scalingo/lua-resty-rollbar/blob/d9bf1c09ec0db0cb81bd4bfd6c1a48306f483d67/lib/resty/rollbar.lua#L170

Hence, if the request to Rollbar fails, nothing wrong happens to the main thread. Or am I missing about what you report?

**Comment by matthieu526-scalingo** [comment]
Closing this as it is more than 3 years old and unlikely to be done as-is. It can always be re-opened if required.
